packages/swing-store/src/transcriptStore.js

```diff
@@ -0,0 +1,443 @@
+   * computed by hashing together the hash from the previous item in its span
+   * together with the new item's own text.
+   *
+   * @param {string} priorHash  The previous item's hash
+   * @param {string} item  The item itself
+   *
+   * @returns {string}  The hash of the combined parameters.
+   */
+  function computeItemHash(priorHash, item) {
+    const itemHash = createSHA256(item).finish();
+    return createSHA256(priorHash).add(itemHash).finish();
+  }
+
+  /**
+   * @type {string} Seed hash to use as the prior hash when computing the hash
+   * of the very first item in a span, since it has no prior item to draw upon.
+   */
+  const initialHash = createSHA256('start of transcript span').finish();
+
+  const sqlWriteSpan = db.prepare(`
+    INSERT INTO transcriptSpans
+      (vatID, startPos, endPos, hash, isCurrent)
+    VALUES (?, ?, ?, ?, ?)
+  `);
+
+  /**
+   * Start a new transcript for a given vat
+   *
+   * @param {string} vatID  The vat whose transcript this shall be
+   */
+  function initTranscript(vatID) {
+    ensureTxn();
+    sqlWriteSpan.run(vatID, 0, 0, initialHash, 1);
+  }
+
+  const sqlGetCurrentSpanBounds = db.prepare(`
+    SELECT startPos, endPos, hash
+    FROM transcriptSpans
+    WHERE vatID = ? AND isCurrent = 1
+  `);
+
+  /**
+   * Obtain the bounds and other metadata for a vat's current transcript span.
+   *
+   * @param {string} vatID  The vat in question
+   *
+   * @returns {{startPos: number, endPos: number, hash: string}}
+   */
+  function getCurrentSpanBounds(vatID) {
+    const bounds = sqlGetCurrentSpanBounds.get(vatID);
+    assert(bounds, `no current transcript for ${vatID}`);
+    return bounds;
+  }
+
+  function spanArtifactName(rec) {
+    return `transcript.${rec.vatID}.${rec.startPos}.${rec.endPos}`;
+  }
+
+  function historicSpanMetadataKey(rec) {
+    if (rec.isCurrent) {
+      return `transcript.${rec.vatID}.current`;
+    } else {
+      return `transcript.${rec.vatID}.${rec.startPos}`;
+    }
+  }
+
+  function currentSpanMetadataKey(rec) {
+    return `transcript.${rec.vatID}.current`;
+  }
+
+  function spanRec(vatID, startPos, endPos, hash, isCurrent) {
+    return { vatID, startPos, endPos, hash, isCurrent };
+  }
+
+  const sqlEndCurrentSpan = db.prepare(`
+    UPDATE transcriptSpans
+    SET isCurrent = 0
+    WHERE isCurrent = 1 AND vatID = ?
+  `);
+
+  /**
+   * End the current transcript span for a vat and start a new one.
+   *
+   * @param {string} vatID  The vat whose transcript is to rollover to a new
+   *    span.
+   */
+  function rolloverSpan(vatID) {
+    ensureTxn();
+    const { hash, startPos, endPos } = getCurrentSpanBounds(vatID);
+    const rec = spanRec(vatID, startPos, endPos, hash);
+    noteExport(historicSpanMetadataKey(rec), JSON.stringify(rec));
+    sqlEndCurrentSpan.run(vatID);
+    sqlWriteSpan.run(vatID, endPos, endPos, initialHash, 1);
+    const newRec = spanRec(vatID, endPos, endPos, initialHash);
```

Comment:
Missing `true` here.
```suggestion
    const newRec = spanRec(vatID, endPos, endPos, initialHash, true);
```

Reply:
No, that's just wrong.  The default spanRec does not have an `isCurrent` value.

Reply:
And... after having written that, it may be that this could be point of divergence between the incremental (callback) feed and the getExportData form, which would obviously be bad.

Reply:
Indeed, that is right.  However, the missing parameter should not be `true` or `false` but `1` or `0`.

Reply:
I'm still confused by the whole historic vs current handling, but yes discrepancy between `noteExport` and `getExportData` is my concern

Reply:
The additional test you proposed (verifying that the incremental feed and the gimme-it-all feed produce the same stuff) would have caught this, so extra evidence in favor of doing that.

Also, debugging this pointed the way to a straightforward path to that test, as there is already a test of the incremental feed that just needs is the addition of an extra call to export followed by a compare.